app/models/setting/auth.rb

```diff
@@ -24,6 +24,7 @@ def self.default_settings
       self.set('authorize_login_delegation', N_("Authorize login delegation with REMOTE_USER environment variable"), false, N_('Authorize login delegation')),
       self.set('authorize_login_delegation_api', N_("Authorize login delegation with REMOTE_USER environment variable for API calls too"), false, N_('Authorize login delegation API')),
       self.set('idle_timeout', N_("Log out idle users after a certain number of minutes"), 60, N_('Idle timeout')),
+      self.set('bcrypt_cost', N_("Cost value of bcrypt password hash function for internal auth-sources (4-30). Higher value is safer but verification is slower particularly for stateless API calls and UI logins. Password change needed to take effect."), 4, N_('BCrypt password cost')),
```

Comment:
I wish we can handle this without adding yet another setting by choosing a reasonable default....

Reply:
Is this a blocker for you, @tbrisker?

Reply:
This should be configurable, I don't think we should be opinionated on security-related things. We actually set a reasonable value during seed, but I'd probably suggest to bump the cost to more safe number like 11 or 12.

I could not reply in the previous thread, so let me reply here:

> wait, isn't the whole idea of using a salt based on it being random? if it is derived from the object, it's not random anymore...

Yes, but for Personal Token the token is actually a random stuff so it does not matter. Honestly I did not investigate why @timogoebel designed it this way, but I can confirm this is safe and I did not want to change behavior in this patch.

Reply:
@tbrisker as much as I would love to provide sane default, re-read our dicussion on discourse about this. Every single person a different view on how the cost should be set. So we default to 50ms processing time during calibration as a reasonably safe but also fast time and we let users to decide.

Reply:
I also disagree that every setting is bad. IMHO settings should have sane defaults, but being able to tweak the app can be a very good thing.